Source/WebCore/style/StyleAdjuster.cpp

```diff
@@ -566,6 +566,16 @@ void Adjuster::adjust(RenderStyle& style) const
             style.setWritingMode(StyleWritingMode::HorizontalTb);
         }
 
+        if (style.gridAutoFlow().direction() == GridAutoFlow::Direction::Normal) {
+            // FIXME: Adjust this once CSSWG clarifies exactly how the initial value should compute on non-masonry.
+            // For now, this gives the most backwards-compatible behavior.
+            if ((style.display() == DisplayType::Masonry || style.display() == DisplayType::InlineMasonry)
```

Comment:
This check is getting pretty hard to follow without careful reading.

Could we do something like:

```
bool isMasonry = style.display() == DisplayType::Masonry || style.display() == DisplayType::InlineMasonry;
bool emptyTemplateColumns = style.gridTemplateColumns().list.isEmpty();
bool emptyTemplateRows = style.gridTemplateRows().list.isEmpty();

if (isMasonry && emptyTemplateColumns && !emptyTemplateRows)
    style.setGridAutoFlowDirection(Style::GridAutoFlow::Direction::Column);
else
    style.setGridAutoFlowDirection(Style::GridAutoFlow::Direction::Row);
```

Reply:
I think this is something we should have more generally -- we need it for Grid and Flexbox, too, and we need it in many places across WebKit -- so my thoughts are to handle it in a follow-up patch to clean up `DisplayType` checks more generally.